lib/tokenlist.cpp

```diff
@@ -706,11 +707,31 @@ static void compileUnaryOp(Token *&tok, AST_state& state, void (*f)(Token *&tok,
     state.op.push(unaryop);
 }
 
+static void skipGenericType(Token *&tok)
+{
+    Token *skip = tok;
+    while (skip && !Token::Match(skip, ",|)")) {
```

Comment:
I prefer a "white-list" instead of "black-list" loop. Loop while tokens are expected.
```suggestion
    while (Token::Match(skip, "%name%|*|:|(")) {
```